vhdx/src/vhdx_header.rs

```diff
@@ -0,0 +1,552 @@
+            Ok(r) => r,
+        };
+
+        let header_1 = Header::new(f, HEADER1_START);
+        let header_2 = Header::new(f, HEADER2_START);
+        let guid: u128 = 0; // TODO: take care of this
+        let (header_1, header_2) = VhdxHeader::update_headers(f, header_1, header_2, guid);
+
+        Ok(VhdxHeader {
+            file_type_identifier,
+            header_1,
+            header_2,
+            region_table_1: RegionTableHeader::new(f, REGION1_START)?,
+            region_table_2: RegionTableHeader::new(f, REGION2_START)?,
+        })
+    }
+
+    fn get_current_header(
+        header_1: Result<Header>,
+        header_2: Result<Header>,
+    ) -> (u8, Result<Header>, Result<Header>) {
+        let mut header1_seq_num: u64 = 0;
+        let mut header2_seq_num: u64 = 0;
+        let mut no_valid_header: bool = true;
+
+        if let Ok(ref header_1) = header_1 {
+            no_valid_header = false;
+            header1_seq_num = header_1.sequence_number;
+        }
+
+        if let Ok(ref header_2) = header_2 {
+            no_valid_header = false;
+            header2_seq_num = header_2.sequence_number;
+        }
+
+        if no_valid_header {
+            panic!("Get current header: {}", Error::NoValidHeader);
+        } else if header1_seq_num >= header2_seq_num {
+            (1, header_1, header_2)
+        } else {
+            (2, header_1, header_2)
+        }
+    }
+
+    /// _update_header() takes two header and update the noncurrent header with
+    /// the corrent one. Returns both headers as a tuple sequenced as it received
+    /// them in the parameter list.
+
+    fn _update_header(
+        f: &mut File,
+        header_1: Result<Header>,
+        header_2: Result<Header>,
+        guid: u128,
+    ) -> (Header, Header) {
+        let (current_header, header_1, header_2) =
+            VhdxHeader::get_current_header(header_1, header_2);
+        if current_header == 1 {
+            match header_1 {
+                Ok(header_1) => {
+                    let header_2 =
+                        Header::update_header(f, header_1.clone(), true, guid, HEADER2_START);
+                    (header_1, header_2)
+                }
+                Err(e) => {
+                    panic!("Header 2 update failed{}", e);
+                }
+            }
+        } else {
+            match header_2 {
+                Ok(header_2) => {
+                    let header_1 =
+                        Header::update_header(f, header_2.clone(), true, guid, HEADER2_START);
+                    (header_1, header_2)
+                }
+                Err(e) => {
+                    panic!("Header 1 update failed{}", e);
+                }
+            }
+        }
+    }
+
+    fn update_headers(
+        f: &mut File,
+        header_1: Result<Header>,
+        header_2: Result<Header>,
+        guid: u128,
+    ) -> (Header, Header) {
+        // According to the spec, update twice
```

Comment:
That's weird, any explanation?? Honestly I don't think this is needed.
And could you please merge `update_headers()` with `_update_header()` function to create only one `update_headers()` one.
Never use `_` as a prefix for a function name.